app/controllers/purchases_controller.rb

```diff
@@ -3,7 +3,12 @@ class PurchasesController < ApplicationController
 
   def index
     @plimit = @@purchase_limit
-    @purchases = Purchase.order(:date).reverse_order.limit(@@purchase_limit)
+    if @@purchase_limit.instance_of? Integer
```

Comment:
could be `if ... elsif ... else` pattern. Maybe extract into private method to make this a bit cleaner?

Reply:
Good idea, and makes it easier to extend in future. Done!